nntrainer/layers/conv2d_layer.cpp

```diff
@@ -143,35 +144,15 @@ static void im2col(const Tensor &in, const TensorDim &kdim,
   unsigned int out_height = (height - eff_k_height) / mstride[0] + 1;
   unsigned int out_width = (width - eff_k_width) / mstride[1] + 1;
 
-  NNTR_THROW_IF(height < eff_k_height, std::invalid_argument)
-    << "calculated height is smaller then effective kernel height."
-    << " padded height: " << height
-    << " kernel height with dilataion: " << eff_k_height
-    << " stride: " << mstride[0];
-
-  NNTR_THROW_IF(width < eff_k_width, std::invalid_argument)
-    << "calculated width is smaller then effective kernel width."
-    << " padded width: " << width
-    << " kernel width with dilataion: " << eff_k_width
-    << " stride: " << mstride[1];
-
-  NNTR_THROW_IF((height - eff_k_height) % mstride[0] != 0,
-                std::invalid_argument)
-    << "calculated height is not integer. "
-    << "padded height: " << height << " kernel height: " << k_height
-    << " stride: " << mstride[0];
-
-  NNTR_THROW_IF((width - eff_k_width) % mstride[1] != 0, std::invalid_argument)
-    << "calculated width is not integer. "
-    << "padded width: " << width << " kernel width: " << k_width
-    << " stride: " << mstride[1];
-
   if (out.uninitialized()) {
-    if (channel_mode) {
-      out = Tensor(out_height * out_width, in.channel() * k_height * k_width);
-    } else {
-      out = Tensor(k_height * k_width, in.channel() * out_height * out_width);
-    }
+    // if (channel_mode) {
```

Comment:
is there any plan to support channel_mode in future?

Reply:
Not for now, we might need channel mode in the near future for the transposed convolution.